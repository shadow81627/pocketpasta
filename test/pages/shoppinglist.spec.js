/**
 * @jest-environment jsdom
 */

import { createLocalVue, mount } from '@vue/test-utils';
import Vuetify from 'vuetify';
import BootstrapVuePlugin from 'bootstrap-vue';
import Component from '@/pages/shoppinglist.vue';

const localVue = createLocalVue();

localVue.use(Vuetify);
localVue.use(BootstrapVuePlugin);

const factory = () =>
  mount(Component, {
    attachToDocument: true,
    // localVue,
  });

describe('shoppinglist page', () => {
  beforeEach(() => {
    document.getSelection = () => {
      return {
        removeAllRanges: () => {},
        getNativeRange: () => {},
        getRangeAt: () => {},
      };
    };
  });

  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
