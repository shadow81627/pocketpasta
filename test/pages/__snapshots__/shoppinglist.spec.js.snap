// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoppinglist page renders properly 1`] = `"<div class=\\"container\\"><div><div id=\\"editor\\"><h1>Shopping list</h1> <br> <h2>Breakfast:</h2> <ul data-checked=\\"false\\"><li>Milk</li> <li>Bananas</li></ul> <br> <h2>Misc:</h2> <ul data-checked=\\"false\\"><li>canola oil</li> <li>apple cider vinegar</li> <li>vinegar</li> <li>baking soda</li> <li>toothpaste</li> <li>toilet paper</li> <li>dish soap</li> <li>snacks</li></ul> <br> <h2>Thursday: DND cookup</h2> <ul data-checked=\\"false\\"><li>macaroni</li> <li>potato</li> <li>carrot</li> <li>broccoli</li> <li>cauliflower</li> <li>kale</li></ul> <br> <h2>Friday: soup leftovers</h2> <ul data-checked=\\"false\\"><li>macaroni</li> <li>potato</li> <li>carrot</li> <li>broccoli</li> <li>cauliflower</li> <li>kale</li></ul> <br> <h2>Saturday: fish and chips</h2> <ul data-checked=\\"false\\"><li>fish</li> <li>chips</li></ul> <br> <h2>Sunday: tacos</h2> <ul data-checked=\\"false\\"><li>tortillas</li> <li>tomato</li> <li>avocado</li> <li>garlic</li> <li>tofu</li> <li>parsley</li></ul> <br> <h2>Monday: Ethiopian</h2> <ul data-checked=\\"false\\"><li>red onion</li> <li>sweet potato</li> <li>walnuts</li> <li>spinach</li> <li>feta</li></ul> <br> <h2>Tuesday: chickpea curry</h2> <ul data-checked=\\"false\\"><li>chickpeas</li> <li>ginger</li> <li>garlic</li> <li>spinach</li></ul> <br> <h2>Wednesday: chickpea curry</h2> <ul data-checked=\\"false\\"><li>chickpeas</li> <li>ginger</li> <li>garlic</li> <li>spinach</li></ul> <br></div> <div class=\\"mt-3\\"><button type=\\"button\\" class=\\"btn btn-secondary\\">Clear</button> <button type=\\"button\\" class=\\"btn float-right btn-primary\\">Save</button></div></div></div>"`;
